src/config.c

```diff
@@ -100,6 +100,37 @@ struct Config *getCommonConfig(void) {
     return &commonConfig;
 }
 
+// Reloads the configuration file and removes interfaces which were removed from config.
+void reloadConfig() {
+    struct vifconfig *OldConfPtr, *TmpConfPtr;
+
+    // Load the new configuration keep reference to the old.
+    OldConfPtr = vifconf;
+    if ( ! loadConfig(configFilePath) ) my_log(LOG_ERR, 0, "reloadConfig: Unable to load config file.");
+
+    // Rebuild the interfaces config.
+    rebuildIfVc();
+
+    my_log(LOG_DEBUG, 0, "reloadConfig: Config Reloaded. OldConfPtr %x, NewConfPtr, %x", OldConfPtr, vifconf);
+    // Free all the old mallocd subnetlists and vifconf list.
+    while ( OldConfPtr ) {
+        TmpConfPtr=OldConfPtr->next; // Increment before free or pointers may be invalid.
+        struct SubnetList *TmpNetPtr;
+        for ( ; OldConfPtr->allowednets;  ) {
```

Comment:
`while`-loop?

Reply:
when you think you have it all covered, you're still doing strange things sometimes. :)